src/syft/lib/pandas/__init__.py

```diff
@@ -0,0 +1,199 @@
+        ("pandas.Series.rdiv", "pandas.Series"),
+        ("pandas.Series.rtruediv", "pandas.Series"),
+        ("pandas.Series.rfloordiv", "pandas.Series"),
+        ("pandas.Series.rmod", "pandas.Series"),
+        ("pandas.Series.rpow", "pandas.Series"),
+        ("pandas.Series.lt", "pandas.Series"),
+        ("pandas.Series.gt", "pandas.Series"),
+        ("pandas.Series.le", "pandas.Series"),
+        ("pandas.Series.ge", "pandas.Series"),
+        ("pandas.Series.ne", "pandas.Series"),
+        ("pandas.Series.eq", "pandas.Series"),
+        ("pandas.Series.argsort", "pandas.Series"),
+        ("pandas.Series.round", "pandas.Series"),
+        ("pandas.Series.head", "pandas.Series"),
+        ("pandas.Series.tail", "pandas.Series"),
+        ("pandas.Series.any", "syft.lib.python.Bool"),
+        ("pandas.Series.shape", "syft.lib.python.Tuple"),
+        ("pandas.Series.all", "syft.lib.python.Bool"),
+        ("pandas.Series.argmax", "syft.lib.python.Int"),
+        ("pandas.Series.nbytes", "syft.lib.python.Int"),
+        ("pandas.Series.mean", "syft.lib.python.Float"),
+        ("pandas.Series.ndim", "syft.lib.python.Int"),
+        ("pandas.Series.size", "syft.lib.python.Int"),
+        ("pandas.Series.hasnans", "syft.lib.python.Bool"),
+        ("pandas.Series.empty", "syft.lib.python.Bool"),
+        ("pandas.Series.T", "pandas.Series"),
+        ("pandas.Series.dropna", "pandas.Series"),
+        ("pandas.Series.to_frame", "pandas.DataFrame"),
+        ("pandas.Series.to_list", "syft.lib.python.List"),
+        (
+            "pandas.Series.sum",
+            UnionGenerator["syft.lib.python.Float", "syft.lib.python.Int"],
+        ),
+        ("pandas.Series.median", "syft.lib.python.Float"),
+        (
+            "pandas.Series.max",
+            UnionGenerator[
+                "syft.lib.python.Bool", "syft.lib.python.Float", "syft.lib.python.Int"
+            ],
+        ),
+        (
+            "pandas.Series.min",
+            UnionGenerator[
+                "syft.lib.python.Bool", "syft.lib.python.Float", "syft.lib.python.Int"
+            ],
+        ),
+    ]
+
+    add_modules(ast, modules)
+    add_classes(ast, classes)
+    add_methods(ast, methods)
+
+    for klass in ast.classes:
+        klass.create_pointer_class()
+        klass.create_send_method()
+        # TODO: Pandas can't have tags and description because they break the dict
+        # klass.create_storable_object_attr_convenience_methods()
+
+    return ast
+
+
+# we cant create Unions that refer to the package itself until the create_ast
+# has completed first so we can call again into post_update_ast to finish these
+# TODO: add support for self referential unions using some kind of post update
+# Issue: https://github.com/OpenMined/PySyft/issues/5323
+def post_create_ast(ast: Globals) -> Globals:
+    self_referencing_methods = [
+        ("pandas.Series.loc", UnionGenerator["pandas.DataFrame", "pandas.Series"]),
```

Comment:
FOOD_FOR_THOUGHT: We might need to create a dependency-like graph for all the libs.

If we continue to add libraries and they have different dependencies (also between them) this might become hell.

Reply:
Yes, I think we need some way to manage this properly.

Reply:
I also feel there is a need for something like this, would definitely love to brain-storm more and work on this 👍